projects/mole/utils.js

```diff
@@ -0,0 +1,206 @@
+
+const decimal_default = require("decimal.js");
+const import_bn6 = require("bn.js");
+
+function asIntN(int, bits = 32) {
+    return Number(BigInt.asIntN(bits, BigInt(int)));
+}
+
+function i32BitsToNumber(v) {
+    return asIntN(BigInt(v), 32);
+}
+
+function toX64_Decimal(num) {
+    return num.mul(decimal_default.pow(2, 64));
+}
+
+function fromX64_Decimal(num) {
+    return num.mul(decimal_default.pow(2, -64));
+}
+
+function signedShiftRight(n0, shiftBy, bitWidth) {
+    const twoN0 = n0.toTwos(bitWidth).shrn(shiftBy);
+    twoN0.imaskn(bitWidth - shiftBy + 1);
+    return twoN0.fromTwos(bitWidth - shiftBy);
+}
+
+function tickIndexToSqrtPricePositive(tick) {
```

Comment:
sorry for delayed review, took a glance earlier, is there some workaround to avoid extra dependency and this function? feel like it can be adjusted a bit (?)

Reply:
Dear @g1nt0ki , thank you for your review. 

We have tried to remove the extra dependency "decimals.js" before, that we put it in our local directory. But this file cannot pass the auto exam. Please look into this previous PR:

https://github.com/DefiLlama/DefiLlama-Adapters/pull/6289/files

So that we have to close it. 

For the tickIndexToSqrtPricePositive function, it is a quick algorithm for pow function. Mole will use Uni V3 CLMM algorithm and calculate the liquidity in CLMM algorithm way.

Reply:
@molefund can you review my code here: https://github.com/DefiLlama/DefiLlama-Adapters/pull/6347/files#diff-2fdc24988439caa562ca0d4e12e2a1c65eca2d83f9bd5b2b6534a99288590482R258

I have refactored the adapter to use the same code we use for uni v3